### PR TITLE
Chore: Update TestData README

### PR DESCRIPTION
**Pull request by andresmgot**
**What is this feature?**

Specifies that TestData is bundled by default.

### CODE REVIEW

**Review by sympatheticmoose — APPROVED** [review]

File: public/app/plugins/datasource/grafana-testdata-datasource/plugin.json [hunk 1]
```diff
@@ -38,6 +38,6 @@
   ],
 
   "dependencies": {
-    "grafanaDependency": ">=5.0.0"
+    "grafanaDependency": ">=10.3.0"
```
Comment (marefr) [inline public/app/plugins/datasource/grafana-testdata-datasource/plugin.json:41]: Hmm not sure this includes Grafana 10.0.3-pre releases? Gcom uses https://semver.npmjs.com/ for semver comparison. I have some vague memory something additional is needed so feel free to check

Comment (andresmgot) [reply public/app/plugins/datasource/grafana-testdata-datasource/plugin.json:41]: It should: https://jubianchi.github.io/semver-check/#/%3E%3D10.3.0/10.3.0-63036

Comment (marefr) [reply public/app/plugins/datasource/grafana-testdata-datasource/plugin.json:41]: Okay great

File: public/app/plugins/datasource/grafana-testdata-datasource/README.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
 # Grafana TestData data source
 
 This data source is used for testing and development of Grafana. Generates test data in different forms.
+
+> **Experimental**: By default, Grafana ships with TestData built in. This behavior can be changed via the `externalCorePlugins` [feature toggle](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/feature-toggles/) and the configuration field `plugin.grafana-testdata-datasource.as_external`. These settings, if enabled, allows you to to install TestData as an external plugin and manage its lifecycle independently of Grafana.
```
Comment (marefr) [inline public/app/plugins/datasource/grafana-testdata-datasource/README.md:5]: As discussed, we can link to https://grafana.com/docs/grafana/latest/datasources/testdata/ and keep the details there.

File: public/app/plugins/datasource/grafana-testdata-datasource/plugin.json (hunk 1 above)
Comment (jackw) [reply public/app/plugins/datasource/grafana-testdata-datasource/plugin.json:41]: Hmm I think it's the go library that we use that has some oddness around semver comparision and prerelease versions... https://pkg.go.dev/github.com/Masterminds/semver/v3#readme-working-with-prerelease-versions

**Comment by andresmgot** [comment]
please @sympatheticmoose take a look again whenever you have a chance to review the wording again :pray:

File: docs/sources/datasources/testdata/_index.md [hunk 3]
```diff
@@ -114,3 +114,15 @@ That makes it much easier for the developers to replicate and solve your issue.
 [panels-visualizations]: "/docs/grafana/ -> /docs/grafana/<GRAFANA VERSION>/panels-visualizations"
 [panels-visualizations]: "/docs/grafana-cloud/ -> /docs/grafana/<GRAFANA VERSION>/panels-visualizations"
 {{% /docs/reference %}}
+
+## Use a custom version of TestData
+
+If you want to use a version of TestData different from the one shipped with Grafana. Note that this procedure is still experimental (requires Grafana 10.3.0). You can do so by following these steps:
```
Comment (sympatheticmoose) [inline docs/sources/datasources/testdata/_index.md:120]:
```suggestion
{{% admonition type="note" %}}
This feature is experimental and requires Grafana version 10.3.0 as a minimum.
{{% /admonition %}}

If you want to use a version of TestData different from the one shipped with Grafana, you can do so by following these steps:
```

File: public/app/plugins/datasource/grafana-testdata-datasource/plugin.json (hunk 1 above)
Comment (andresmgot) [reply public/app/plugins/datasource/grafana-testdata-datasource/plugin.json]: it seems that `gen-cue` doesn't like pre releases :roll_eyes:

**Review by xnyo — APPROVED** [review]
LGTM!

File: docs/sources/datasources/testdata/_index.md [hunk 4]
```diff
@@ -114,3 +114,19 @@ That makes it much easier for the developers to replicate and solve your issue.
...
+## Use a custom version of TestData
+
+{{% admonition type="note" %}}
+This feature is experimental and requires Grafana version 10.3.0 as a minimum.
+{{% /admonition %}}
+
+If you want to use a version of TestData different from the one shipped with Grafana, you can do so by following these steps:
+
+1. Enable the [feature toggle](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/feature-toggles/) `externalCorePlugins`.
+1. Set the configuration field `plugin.grafana-testdata-datasource.as_external` to `true`.
```
Comment (wbrowne) [inline docs/sources/datasources/testdata/_index.md:127]: I wonder is this explicit enough for people to know what to put in the `ini`? Just not seen it before described this way

Comment (andresmgot) [reply docs/sources/datasources/testdata/_index.md:127]: fair point :+1:

**Review by sympatheticmoose — APPROVED** [review]
words look good to me 🙂

File: docs/sources/datasources/testdata/_index.md [hunk 5]
```diff
@@ -114,3 +114,28 @@ That makes it much easier for the developers to replicate and solve your issue.
...
+   [feature_toggles]
+   externalCorePlugins = true
+
+   [plugin.grafana-testdata-datasource]
+   as_external = true
+   ```
+
+1. Restart Grafana.
+
+These settings, if enabled, allows you to to install TestData as an external plugin and manage its lifecycle independently of Grafana.
```
Comment (josmperez) [inline docs/sources/datasources/testdata/_index.md:139]:
```suggestion
These settings, if enabled, allow you to to install TestData as an external plugin and manage its lifecycle independently of Grafana.
```

File: public/app/plugins/datasource/grafana-testdata-datasource/README.md [hunk 6]
```diff
@@ -1,3 +1,9 @@
 # Grafana TestData data source
 
 This data source is used for testing and development of Grafana. Generates test data in different forms.
+
+Grafana ships with **built in** support for TestData. You just have to add it as a data source and you will be ready to build dashboards.
```
Comment (josmperez) [inline public/app/plugins/datasource/grafana-testdata-datasource/README.md:5]:
```suggestion
Grafana ships with _built-in_ support for TestData. You just have to add it as a data source and you will be ready to build dashboards.
```

File: docs/sources/datasources/testdata/_index.md [hunk 7]
```diff
@@ -114,3 +114,28 @@ That makes it much easier for the developers to replicate and solve your issue.
 [panels-visualizations]: "/docs/grafana/ -> /docs/grafana/<GRAFANA VERSION>/panels-visualizations"
 [panels-visualizations]: "/docs/grafana-cloud/ -> /docs/grafana/<GRAFANA VERSION>/panels-visualizations"
 {{% /docs/reference %}}
+
+## Use a custom version of TestData
+
+{{% admonition type="note" %}}
+This feature is experimental and requires Grafana version 10.3.0 as a minimum.
```
Comment (josmperez) [inline docs/sources/datasources/testdata/_index.md:121]:
```suggestion
This feature is experimental and requires Grafana version 10.3.0 or later.
```

File: docs/sources/datasources/testdata/_index.md [hunk 8]
```diff
@@ -114,3 +114,28 @@ That makes it much easier for the developers to replicate and solve your issue.
...
+{{% /admonition %}}
+
+If you want to use a version of TestData different from the one shipped with Grafana, you can do so by following these steps:
```
Comment (josmperez) [inline docs/sources/datasources/testdata/_index.md:124]:
```suggestion
If you want to use a version of TestData different from the one shipped with Grafana, follow these steps:
```

**Review by josmperez — APPROVED** [review]
A few nits, otherwise LGTM.

**Review by wbrowne — APPROVED** [review]
LGTM

**Pull request merged**